src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs

```diff
@@ -375,10 +375,12 @@ internal static bool IsClosureScope(SyntaxNode node)
                 // With the introduction of pattern-matching, many nodes now contain top-level
                 // expressions that may introduce pattern variables.
                 case SyntaxKind.EqualsValueClause:
-                case SyntaxKind.CatchFilterClause:
```

Comment:
Why is `CatchFilterClause` removed? Are the pattern variables introduced in filter clauses subsumed in another `SyntaxKind`?